File: longestSubsequence.py
from typing import List
def longestSubs(a: List[int]) -> int:
    n = len(a)
    longest = 0
    st = set()
    
    for i in range(n):
        st.add(a[i])
    
    for it in st:
        if it - 1 not in st:
            cnt = 1
            x = it
            while x + 1 in st:
                x += 1
                cnt += 1
            longest = max(longest, cnt)
    return longest

if __name__ == "__main__":
    a = [100, 200, 1, 2, 3, 4, 0]
    ans = longestSubs(a)
    print("The longest subsequence is ", ans)
    

File: longestSubsequence.cpp
#include <iostream>
#include <vector>
#include <unordered_set>

using namespace std;

int longest_subsequence(vector<int> &a)
{
    int n = a.size();
    int longest = 0;

    unordered_set<int> st;
    for (auto it : a)
    {
        st.insert(it);
    }
    for (auto it : st)
    {
        if (st.find(it - 1) == st.end())
        {
            int cnt = 1;
            int x = it;
            while (st.find(x + 1) != st.end())
            {
                x++, cnt++;
            }
            longest = max(longest, cnt);
        }
    }
    return longest;
}

int main()
{
    vector<int> a = {100, 200, 1, 2, 4, 3, 0};
    cout << "The longest successive subsequence is " << longest_subsequence(a) << endl;
    return 0;
}